### ISSUE TITLE
refactor: remove .distinct method from ColumnExpr

### ISSUE THREAD

**Issue by toryhaavik** [ibis-project/ibis#1711]
Given an expression like:
`expr[expr.column.distinct().name('thing'), ibis.literal(1).name('dummy')]`
you'd expect to have a table with N rows per distinct values in `expr.column`, each with a value of `1` in the `dummy` column.

This is similar to `SELECT DISTINCT column, 1 AS 'dummy' FROM table` in SQL.

With the pandas backend, you get the same number of rows as `expr` would generate, with `nan`s filled in the column after the distinct values.

**Comment by cpcloud** [comment]
So, I implemented a "fix" for this which involved adding a an analysis pass and spending some time thinking about the `DistinctColumn` operation.

I'm now really against keeping this operation around as leads to expressions that are very confusing to read and whose semantics are difficult to reason about, negating any convenience provided by having this method. Moreover, no backends appear to implements this operation correctly

Here's an expression that would make me raise an eyebrow if I came across it:

```python
t[2 * t.a.distinct() + t.b.distinct().log().distinct() + t.c.log().distinct()]
```

I'd like to propose that we deprecate the `distinct()` method on columns. Any correct behavior that was previously possible can be done with a combination of `TableExpr.distinct()` and projections.

@toryhaavik Any thoughts on this? Making this change would move this issue into major release territory, if that's helpful to know.
